src/common/useOutsideEvent.ts

```diff
@@ -1,10 +1,15 @@
 import * as React from 'react';
 
 // Utilizes the useEffect hook to handle when an event is fired on the document, but outside of the specified element
-export default function useOutsideEvent(type: string, ref?: React.RefObject<Element>, callback?: () => void): void {
+export default function useOutsideEvent(
+    type: string,
+    ref?: React.RefObject<Element> | Element | null,
```

Comment:
I tend to believe the current implementation is actually flawed, since the ref itself will not change between renders. Can we change this to accept an element only?